popde/density_estimate.py

```diff
@@ -1,3 +1,83 @@
+import numpy as np
+
+class SimpleKernelDensityEstimation:
+    """
+    Fit and evaluate multi-dimensional Kernel Density Estimation (KDE) 
+
+    Methods:
+    --------
+    check_dimensionality():
+        check if the data matches the dimensions of KDE.
+
+    evaluate(points):
+        Evaluate the KDE at given data points.
+
+    Examples:
+    ---------
+    # Create a KDE instance and fit it to data
+    data = np.random.randn((100, 100))
+    kde = SimpleGaussianKernelDensityEstimation(data)
+
+    # Evaluate the KDE at new data points
+    new_data = np.array(np.linspace(-3, 3, 100).tolist(), np.linspace(-3, 3, 100).tolist())
+    density_values = kde.evaluate(new_data)
+    """
+    def __init__(self, data, bandwidth=1.0, kernel='gaussian', dim_names=None):
+        """
+        Initialize the KernelDensityEstimation object.
+        data: array-like, shape (n_samples, n_features)
+               points of the data define each kernel position
+               each row is a point, eachcolumn is a feature.
+        kwargs:
+            dim_names : sequence of dimension names, e.g. ('m1', 'z', 'chi_eff') 
+                        values must be strings
+        """
+        if len(data.shape) != 2:
+            raise ValueError("data must have shape (n_samples, n_features).")
+
+        self.data = np.asarray(data) 
+        self.bandwidth = bandwidth
+        self.kernel = kernel 
+
+        self.dim_names = dim_names
+        if dim_names is not None:
+            self.check_dimensionality()
+
+   def check_dimensionality(self):
+       """
+       Check if the dimension of KDE matches the parameters on which we evaluate KDE.
+        Raises ValueError
+       """
+        if len(self.data.shape[0]) != len(self.dim_names):
+            raise ValueError("Dimensionality of data array does not match the number of dimension names.")
+
+
+    def evaluate(self, points):
+        """
+        Evaluate the KDE at given data points.
+
+        Parameters:
+        -----------
+        points : array-like
+            The data points at which the KDE will be evaluated.
+
+        Returns:
+        --------
+        density_values : array-like
+            The estimated density values at the given data points.
+            using standard gaussian distribution
+        """
+        import scipy
+        from scipy.stats import gaussian_kde 
+
+        points = np.asarray(points).T
+        kernel_function = gaussian_kde(self.data.T, bw_method=self.bandwidth)
```

Comment:
So this initializes a scipy object which can be used to evaluate the KDE later if it is set as an attribute of the SimpleKernelDensityEstimation object, ie we could make the kernel function once and effectively save it. 
I guess we can consider this in a later improvement